File: FLIR/01_Acquisition.py
#!/usr/bin/env python3

"""
Shows how to acquire an image. See Acquisition.py example
Date: 2019-12-29
"""

import PySpin


system = PySpin.System_GetInstance()

cam_list = system.GetCameras()
cam = cam_list.GetByIndex(0)
cam.Init()

print('*** IMAGE ACQUISITION ***\n')
cam.AcquisitionMode.SetValue(PySpin.AcquisitionMode_Continuous)
print('Acquisition mode set to continuous...')

cam.BeginAcquisition()
cam_serial_number = cam.DeviceSerialNumber.ToString()
NUM_IMAGES = 10
for i in range(NUM_IMAGES):
    img = cam.GetNextImage()
    if img.IsIncomplete():
        print('Image incomplete with image status %d ...' % img.GetImageStatus())
    else:
        width = img.GetWidth()
        height = img.GetHeight()
        print('Grabbed Image %d, width = %d, height = %d' % (i, width, height))
        img_converted = img.Convert(PySpin.PixelFormat_Mono8, PySpin.HQ_LINEAR)
        filename = 'Acquisition-{}-{:02}.jpg'.format(cam_serial_number, i)
        img_converted.Save(filename)
        print('Image saved at {}'.format(filename))
        # release in order to keep from filling the buffer
        img.Release()
cam.EndAcquisition()

cam.DeInit()
del cam
cam_list.Clear()
system.ReleaseInstance()

File: FLIR/01_Acquisition.cpp
/*
 * Shows how to acquire an image.
 *
 * */


#include "Spinnaker.h"
#include <iostream>

using namespace Spinnaker;
using namespace Spinnaker::GenApi;
using namespace Spinnaker::GenICam;

int main(int /*argc*/, char** /*argv*/)
{
    std::cout << "FLIR: Acquisition\n";
    auto system = System::GetInstance();
    auto cam_list = system->GetCameras();

    if (cam_list.GetSize() == 0) {
        std::cerr << "No cameras found\n";
        cam_list.Clear();
        system->ReleaseInstance();
        return -1;
    }

    auto cam = cam_list.GetByIndex(0);
    cam->Init();
    std::cout << "Fancy processing\n";
    cam->DeInit();

    printf("*** IMAGE ACQUISITION ***\n");
    cam->AcquisitionMode.SetValue(AcquisitionMode_Continuous);
    printf("Acquisition mode set to continuous...");

    cam->BeginAcquisition();
    auto cam_serial_number = cam->DeviceSerialNumber.ToString();
    printf("SN = %s\n", cam_serial_number.c_str());
    const int NUM_IMAGES = 10;
    for (int i = 0; i < NUM_IMAGES; i++) {
        auto img = cam->GetNextImage();
        if (img->IsIncomplete()) {
            fprintf(stderr, "Image incomplete with image statue %d ...", img->GetImageStatus());
        }
        else {
            auto width = img->GetWidth();
            auto height = img->GetHeight();
            printf("Grabbed Image %d, width = %lu, height = %lu", i, width, height);
            auto img_converted = img->Convert(PixelFormat_Mono8, HQ_LINEAR);
            char filename[100];
            sprintf(filename, "Acquisition-{%s}-{%02d}.jpg", cam_serial_number.c_str(), i);
            img_converted->Save(filename);
            img->Release();
        }
    }
    cam->EndAcquisition();

    // Release reference to the camera
    cam = nullptr;
    cam_list.Clear();
    system->ReleaseInstance();
    return 0;
}

